src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -297,6 +305,10 @@ public function getName()
             if (ctype_digit($this->name[0])) {
                 $this->name = '_'.$this->name;
             }
+
+            if ('app' !== $this->name) {
```

Comment:
Why `app` here? Why does that indicate that it's been auto-discovered?

And what would this look like in 4.0? Would this method exist... but throw an exception? (so that the user either needs to override this method or pass in as an arg?)

Reply:
For 4.0 auto-discovery is removed and it should look like a simple getter.

Reply:
@weaverryan AFAIK, in 4.0, this method would be `getName() { return 'app';}`. So anyone getting a different name through auto-discovery will need to override the method instead, to see the name explicitly (and so this code would not run anymore).
If the auto-discovered name is `app` (which is the case for all standard-edition projects), migrating to 4.0 would not break things as the hardcoded `app` would do the same.

We have 2 solutions in 4.0:
- make the method return an hardcoded name and tell multi-kernel projects that they should override the method to avoid bugs. Existing multi-kernel projects will be warned for this by the deprecation. And new multi-kernels project should be warned about it with a big warning in the 4.0 documentation about multi-kernel setup
- make the method abstract. This is a bigger BC break as all SE projects would have to implement the method, but it would make the explanation simpler for multi-kernel projects as they would work the same (everyone has to overwrite the method). For flex projects, this is not an issue as the recipe is doing the override already (to have it named `app` and not `src` in 3.x projects), so it would just mean keeping the override in the 4.0 recipe.

If we go for the second option, the deprecation should instead be moved at the very beginning of `getName` and say `Not overriding Kernel::getName` is deprecated since Symfony 3.4. The method will be astract in 4.0.`

Reply:
@stof for 4.0 it should look like `getName() { return $this->name; }`